### ISSUE TITLE
Example App: Adding cleanup to the setup hook causes error

### ISSUE THREAD

**Issue by andrekovac** [react-native-kit/react-native-track-player#930]
**Describe the bug**

I want to stop the player when leaving the player screen.

But I get following error message when adding a cleanup return function and I then leave the player screen and then return back to it.

```
Error: You attempted to set the key `0` with the value `{"id":"1111","url":"https://drive.google.com/uc?export=download&id=1AjPwylDJgR8DOnmJWeRgZzjsohi-7ekj","title":"Longing","artist":"David Chavez","artwork":"https://picsum.photos/200","duration":143}` on an object that is meant to be immutable and has been frozen.
```

**To Reproduce**

1. Change https://github.com/react-native-kit/react-native-track-player/blob/dev/example/react/screens/PlaylistScreen.js in the following way:

```diff
  useEffect(() => {
    setup();
+    return () => {
+      exitPlayer();
    };
  }, []);

+  const exitPlayer = async () => {
+    try {
+      await TrackPlayer.stop();
+    } catch (error) {
+      console.error('exitPlayer', error);
+    }
+  };
```

2. Open the app in the iOS Simulator, tap on "Playlist Example" to get to the player. Hit "Play" and listen to the song for 5 seconds. Then tap on "Back" to leave the player.
3. Tap on "Playlist Example" again to get to the player and hit "Play" again. --> The above described error appears.

Adding `TrackPlayer.destroy();` to the `exitPlayer` function doesn't help.

**Environment (please complete the following information):**
Run `react-native info` in your project and share the content.

```
  React Native Environment Info:
    System:
      OS: macOS 10.14.6
      CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
      Memory: 66.66 MB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.14.1 - ~/.nvm/versions/node/v10.14.1/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 6.14.4 - ~/.nvm/versions/node/v10.14.1/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
      Android SDK:
        API Levels: 23, 24, 25, 26, 27, 28
        Build Tools: 23.0.1, 24.0.2, 25.0.0, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 27.0.2, 27.0.3, 28.0.2, 28.0.3, 29.0.0, 29.0.0, 29.0.0, 29.0.2, 30.0.0, 30.0.0
        System Images: android-23 | Intel x86 Atom_64, android-23 | Google APIs Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 3.6 AI-192.7142.36.36.6308749
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.3 => 16.8.3
      react-native: 0.59.9 => 0.59.9
    npmGlobalPackages:
      create-react-native-module: 0.18.0
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7
```

What `react-native-track-player` version are you using?

- It's the version in the example app, so it's `file:..` (last commit: `98a54be1c58a801f715a269f5d387ce729083930` from `Tue May 5 09:33:57 2020 -060`) 
The same error also occurred on an earlier version with the latest commit from April 28.

Are you testing on a real device or in the simulator? Which OS version are you running?

Tested on iOS Simulator iPhone 11 - iOS 13.3

**Code**

I tested it in the example app so its easy to reproduce and described the changes above.

**Comment by puckey** [comment]
Related to https://github.com/react-native-kit/react-native-track-player/issues/742

**Comment by curiousdustin** [comment]
Fix released in [1.2.4](https://github.com/react-native-kit/react-native-track-player/releases/tag/v1.2.4)
